protocol/messenger_profile_showcase.go

```diff
@@ -1,68 +1,83 @@
 package protocol
 
 import (
+	"bytes"
 	"crypto/ecdsa"
 	crand "crypto/rand"
 	"errors"
 	"reflect"
+	"sort"
 
 	"github.com/golang/protobuf/proto"
+	"go.uber.org/zap"
 
+	"github.com/status-im/status-go/eth-node/crypto"
 	"github.com/status-im/status-go/multiaccounts/accounts"
 	"github.com/status-im/status-go/protocol/common"
 	"github.com/status-im/status-go/protocol/communities"
+	"github.com/status-im/status-go/protocol/identity"
 	"github.com/status-im/status-go/protocol/protobuf"
 )
 
-var errorNoAccountProvidedWithTokenOrCollectible = errors.New("no account provided with tokens or collectible")
-var errorDublicateAccountAddress = errors.New("duplicate account address")
-
-// NOTE: this error is temporary unused because we don't know account on this stage
-// var errorNoAccountAddressForCollectible = errors.New("no account found for collectible")
-var errorAccountVisibilityLowerThanCollectible = errors.New("account visibility lower than collectible")
 var errorDecryptingPayloadEncryptionKey = errors.New("decrypting the payload encryption key resulted in no error and a nil key")
 
-func toProfileShowcaseCommunityProto(preferences []*ProfileShowcaseCommunityPreference, visibility ProfileShowcaseVisibility) []*protobuf.ProfileShowcaseCommunity {
-	communities := []*protobuf.ProfileShowcaseCommunity{}
+func (m *Messenger) toProfileShowcaseCommunityProto(preferences []*identity.ProfileShowcaseCommunityPreference, visibility identity.ProfileShowcaseVisibility) []*protobuf.ProfileShowcaseCommunity {
+	entries := []*protobuf.ProfileShowcaseCommunity{}
 	for _, preference := range preferences {
 		if preference.ShowcaseVisibility != visibility {
 			continue
 		}
 
-		communities = append(communities, &protobuf.ProfileShowcaseCommunity{
+		entry := &protobuf.ProfileShowcaseCommunity{
 			CommunityId: preference.CommunityID,
 			Order:       uint32(preference.Order),
-		})
+		}
+
+		community, err := m.communitiesManager.GetByIDString(preference.CommunityID)
+		if err != nil {
+			m.logger.Warn("failed to get community for profile entry ", zap.Error(err))
+		}
+
+		if community != nil && community.Encrypted() {
```

Comment:
Should we return an error if `community == nil`? That would mean the user tried to send a community they don't even know?

Reply:
If an error occurred, the community can be nil here. The error does not break the loop and is simply logged. I could do a stricter check and return an error, but I think the profile showcase should be more tolerant of missing data.

Reply:
@jrainville  `getByID` has changed recently, it returns error in case there is no community, instead of nil community and no error -> https://github.com/status-im/status-go/pull/4544